### DISCUSSION TITLE
Could u please tell me how to make Comfyui to use all cpu's threads?

### DISCUSSION THREAD

**Discussion by Amin456789** [Comfy-Org/ComfyUI#1634] [Q&A]
Hello!
first of all thank you for this
can somebody tell me please how to make comfyui to use all my cpu threads?
in webui somebody said we can change it by:
https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/3387

torch.set_num_threads(<number of threads>)
torch.set_num_interop_threads(<number of threads>)

but i don't know where i can change these in comfyui.

kind regards

**Comment by drago87** [comment]
i would like to do the same. or if there is a command that makes it run on an AMD GPU

**Reply by rafa-br34** [reply]
same

**Comment by Digitus78** [comment]
comfy/utils.py looks the easiest to set right before a load.  Might want to do a few torch performance tests to see how much it will actually slow down not speed up.   Intel has done a good job at tuning their chips for torch.  

util already has torch imported so no need to import

set_torch_threads(2)
set_torch_interop_threads(2)

would do a 

def cpu_torch_threads()

if multiprossesor.cpu_count() == get_torch_threads()
print say something cool 

   else

  set_torch_threads("your multiprocessor count^^")

no that is not correct syntax but its a idea.  Then have it reset at shut down.
find a base line otherwise its counter productive.

**Reply by rafa-br34** [reply]
Yep, I had used set_torch_threads on comfy/model_management.py to test it but for some reason, it does indeed slow down, however, I'm using a POWER8 machine so it doesn't make much sense why it would decrease performance. How can that be??
I don't think NUMA causes it since I tried without but it still didn't help. I'd guess it's PyTorch somehow misusing the threads, but that wouldn't make much sense.
